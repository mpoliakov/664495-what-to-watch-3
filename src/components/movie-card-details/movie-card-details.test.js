import React from 'react';
import renderer from 'react-test-renderer';
import MovieCardDetails from './movie-card-details.jsx';

it(`<MovieCardDetails/> is rendered correctly`, () => {
  const film = {
    name: `Aviator`,
    posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Aviator.jpg`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/aviator.jpg`,
    backgroundImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Aviator.jpg`,
    backgroundColor: `#D6CDAF`,
    description: `A biopic depicting the early years of legendary Director and aviator Howard Hughes' career from the late 1920s to the mid 1940s.`,
    rating: 9.8,
    scoresCount: 307174,
    director: `Martin Scorsese`,
    starring: [
      `Leonardo DiCaprio`,
      `Cate Blanchett`,
      `Kate Beckinsale`
    ],
    runTime: 170,
    genre: `Drama`,
    released: 2014,
    id: 1,
    isFavorite: false,
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  };

  const component = renderer.create(<MovieCardDetails film={film} />).toJSON();
  expect(component).toMatchSnapshot();
});
