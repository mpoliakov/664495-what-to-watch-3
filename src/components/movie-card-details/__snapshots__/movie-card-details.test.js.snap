// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCardDetails/> is rendered correctly 1`] = `
<div
  className="movie-card__text movie-card__row"
>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Director
      </strong>
      <span
        className="movie-card__details-value"
      >
        Martin Scorsese
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Starring
      </strong>
      <span
        className="movie-card__details-value"
      >
        Leonardo DiCaprio
        , 
        <br />
        Cate Blanchett
        , 
        <br />
        Kate Beckinsale
      </span>
    </p>
  </div>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Run Time
      </strong>
      <span
        className="movie-card__details-value"
      >
        2h 50m
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Genre
      </strong>
      <span
        className="movie-card__details-value"
      >
        Drama
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Released
      </strong>
      <span
        className="movie-card__details-value"
      >
        2014
      </span>
    </p>
  </div>
</div>
`;
